app/controllers/cart_items_controller.rb

```diff
@@ -0,0 +1,37 @@
+class CartItemsController < ApplicationController
+  def new
+    @cart_item = CartItem.new
+  end
+
+  # DELETE /cart_item/1
+  def destroy
+    @cart_item = CartItem.find(params[:id])
+    @cart_item.destroy
+    respond_to do |format|
+      format.html { redirect_to '/cart', notice: 'The item was successfully removed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def add_to_cart
+    @cart = current_user.cart
+    @cart_item = CartItem.new(cart_id: @cart.id, product_id: params[:product_id], amount: params[:amount])
```

Comment:
Q1

No validation of `product_id` or `amount`. `product_id` may name an ID for which no product exists. `amount` may be zero or even negative.